lib/chatbot-api/appsync-ws.ts

```diff
@@ -0,0 +1,133 @@
+import * as cdk from "aws-cdk-lib";
+import * as appsync from "aws-cdk-lib/aws-appsync";
+import { Code, Function, LayerVersion, Runtime } from "aws-cdk-lib/aws-lambda";
+import { SqsEventSource } from "aws-cdk-lib/aws-lambda-event-sources";
+import { Construct } from "constructs";
+import { Shared } from "../shared";
+import { IQueue } from "aws-cdk-lib/aws-sqs";
+import { ITopic } from "aws-cdk-lib/aws-sns";
+import { UserPool } from "aws-cdk-lib/aws-cognito";
+import { NodejsFunction } from "aws-cdk-lib/aws-lambda-nodejs";
+import * as path from "path";
+
+interface ChatGraphqlApiProps {
+  readonly queue: IQueue;
+  readonly topic: ITopic;
+  readonly userPool: UserPool;
+  readonly shared: Shared;
+}
+
+export class ChatGraphqlApi extends Construct {
+  public readonly apiKey: string | undefined;
+  public readonly graphQLUrl: string | undefined;
+  public readonly graphQLApi: appsync.GraphqlApi;
+  public readonly outgoingMessageHandler: Function;
+
+  constructor(scope: Construct, id: string, props: ChatGraphqlApiProps) {
+    super(scope, id);
+
+    const powertoolsLayerJS = LayerVersion.fromLayerVersionArn(
+      this,
+      "PowertoolsLayerJS",
+      `arn:aws:lambda:${
+        cdk.Stack.of(this).region
+      }:094274105915:layer:AWSLambdaPowertoolsTypeScript:22`
+    );
+
+    // makes a GraphQL API
+    const api = new appsync.GraphqlApi(this, "ws-api", {
+      name: "chatbot-ws-api",
+      schema: appsync.SchemaFile.fromAsset(
+        "lib/chatbot-api/schema/schema-ws.graphql"
+      ),
+      authorizationConfig: {
+        additionalAuthorizationModes: [
+          {
+            authorizationType: appsync.AuthorizationType.IAM,
+          },
+          {
+            authorizationType: appsync.AuthorizationType.USER_POOL,
+            userPoolConfig: {
+              userPool: props.userPool,
+            },
+          },
+        ],
+      },
+      xrayEnabled: true,
+    });
+
```

Comment:
Same issue as `rest-api.ts` 

This introduces the following deprecation warning on synthesis:
```
[WARNING] aws-cdk-lib.aws_appsync.GraphqlApiProps#schema is deprecated.
  use Definition.schema instead
  This API will be removed in the next major release.
```

Can this be resolved to avoid introducing calls to deprecated code?